### PR TITLE
Human language interpreter

### PR DESCRIPTION
**Pull request by tilmankamp**
Signed-off-by: Tilman Kamp <tilmankamp@posteo.de>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Thanks for the PR!
A few comments upfront:
- Note that we discussed in https://github.com/eclipse/smarthome/issues/584#issuecomment-185131143 to use `io.voice.interpreter` as a package. I think we should actually remove the current `io.multimedia` bundle and introduce a `io.voice` and `io.audio` instead. 
- The console command should then also be added to the `io.voice` bundle and not have it included in the `io.console` bundle.

**Comment by tilmankamp** [comment]
@kaikreuzer : Got it - I'll introduce the `io.voice.interpreter` package in a new bundle `io.voice` as part of this PR. However: I won't remove/rename `io.multimedia` - instead I'll just move parts of it into `io.voice`. I think it's better, if the contributor of the new `io.audio` does the `io.multimedia` to `io.audio` bundle refactoring.

**Comment by kaikreuzer** [comment]
Yep, makes sense!

**Comment by tilmankamp** [comment]
@kaikreuzer : Now there is no `multimedia` bundle anymore. All content was about TTS, so I decided to just refactor it to `voice`. This new `voice` bundle now also contains the `say` console command.

File: bundles/io/org.eclipse.smarthome.io.voice/src/main/java/org/eclipse/smarthome/io/voice/internal/text/StandardHumanLanguageInterpreter.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+ */
+public class StandardHumanLanguageInterpreter implements HumanLanguageInterpreter {
+
+    private ItemRegistry itemRegistry;
+
+    private EventPublisher eventPublisher;
+
+    private final Set<Locale> supportedLocales = Collections.unmodifiableSet(Collections.singleton(Locale.ENGLISH));
+
+    private final String PADON = "I didn't quite follow that.";
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.voice/src/main/java/org/eclipse/smarthome/io/voice/internal/text/StandardHumanLanguageInterpreter.java]: typo: PARDON

File: bundles/io/org.eclipse.smarthome.io.voice/src/main/java/org/eclipse/smarthome/io/voice/internal/extensions/InterpretConsoleCommandExtension.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        BundleContext context = FrameworkUtil.getBundle(this.getClass()).getBundleContext();
+        Collection<ServiceReference<HumanLanguageInterpreter>> refs = null;
+        try {
+            refs = context.getServiceReferences(HumanLanguageInterpreter.class, null);
+        } catch (InvalidSyntaxException e) {
+            // should never happen
+        }
+        if (refs != null && refs.size() > 0) {
+            try {
+                console.println(context.getService(refs.iterator().next()).interpret(Locale.ENGLISH, msg));
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.voice/src/main/java/org/eclipse/smarthome/io/voice/internal/extensions/InterpretConsoleCommandExtension.java]: I would expect the system locale to be used here.

**Comment by kaikreuzer** [comment]
Thanks @tilmankamp, looks very good to me!
I think for having the build working, you will also need to adapt the [feature](https://github.com/eclipse/smarthome/blob/master/features/org.eclipse.smarthome.feature.runtime.core/feature.xml#L51) since the bundle name has changed.
Is the `StandardHumanLanguageInterpreter` meant as a demo implementation or as a very initial version of a fully-fledged interpreter yet to come?

**Comment by tilmankamp** [comment]
@kaikreuzer : The language is now picked from the supported ones of the interpreter. Feature got renamed. Typos fixed.

**Comment by tilmankamp** [comment]
@kaikreuzer 
Regarding your question:
> Is the StandardHumanLanguageInterpreter meant as a demo implementation or as a very initial version of a fully-fledged interpreter yet to come?

The idea was to provide a standard interpreter that will support a handful of essential commands. Like 

- turn thing on/off
- open/close thing
- set thing to value [percent]
- get status of thing
- [...]

So - yes - this will get further extended. What do you think about this?

**Comment by kaikreuzer** [comment]
> So - yes - this will get further extended. What do you think about this?

Ok for me for a start. I am not clear yet though whether this is supposed to be THE interpreter on the long run, so we enhance it continuously and it is our reference implementation, or if this is just a simple and constrained implementation, which we might later actually want to separate out into a bundle on its own, so that it can be easily replaced by a different, more powerful implementation.

Thanks!

**Comment by kaikreuzer** [comment]
Argh, sorry, I completely missed to ask you to squash before I merged it :disappointed:

**Comment by tilmankamp** [comment]
Will/Could you squash on master?

**Comment by kaikreuzer** [comment]
No, I don't do any history rewrite on master - if @maggu2810 does not beat me up, I will suggest to leave the commits just in there as they are now...

**Comment by tilmankamp** [comment]
The pom.xml also needs an update.

**Comment by maggu2810** [comment]
> @maggu2810 does not beat me up

I don't :wink:

**Pull request merged**